apps/api/src/app/environments-v1/usecases/output-renderers/in-app-output-renderer.usecase.ts

```diff
@@ -1,87 +1,57 @@
-    const inApp = InAppRenderOutputSchema.optional().parse(renderCommand.controlValues);
+    const inApp: InAppControlType = InAppControlZodSchema.parse(renderCommand.controlValues);
     if (!inApp) {
       throw new Error('Invalid in-app control value data');
     }
 
+    const { primaryAction, secondaryAction, redirect } = inApp;
+
     return {
       subject: inApp.subject,
       body: inApp.body,
       avatar: inApp.avatar,
-      primaryAction:
-        inApp.primaryAction &&
-        inApp.primaryAction.label &&
-        inApp.primaryAction.redirect &&
-        inApp.primaryAction.redirect.url
-          ? {
-              label: inApp.primaryAction.label,
-              redirect: {
-                url: inApp.primaryAction.redirect.url,
-                target: inApp.primaryAction.redirect.target as RedirectTargetEnum,
-              },
-            }
-          : undefined,
-      secondaryAction:
-        inApp.secondaryAction &&
-        inApp.secondaryAction.label &&
-        inApp.secondaryAction.redirect &&
-        inApp.secondaryAction.redirect.url
-          ? {
-              label: inApp.secondaryAction?.label,
-              redirect: {
-                url: inApp.secondaryAction?.redirect.url,
-                target: inApp.secondaryAction?.redirect.target as RedirectTargetEnum,
-              },
-            }
-          : undefined,
-      redirect:
-        inApp.redirect && inApp.redirect.url
-          ? {
-              url: inApp.redirect.url,
-              target: inApp.redirect.target as RedirectTargetEnum,
-            }
-          : undefined,
+      primaryAction: this.buildActionIfAllPartsAvailable(primaryAction),
+      secondaryAction: this.buildActionIfAllPartsAvailable(secondaryAction),
+      redirect: this.buildRedirect(redirect),
       data: inApp.data as Record<string, unknown>,
     };
   }
-}
-const RedirectTargetEnumSchema = z.enum(['_self', '_blank', '_parent', '_top', '_unfencedTop']);
 
-const InAppRenderOutputSchema = z.object({
-  subject: z.string().optional(),
-  body: z.string(),
-  avatar: z.string().optional(),
-  primaryAction: z
-    .object({
-      label: z.string().optional(),
-      redirect: z.object({
-        url: z.string().optional(),
-        target: RedirectTargetEnumSchema.optional(), // Optional target
-      }),
-    })
-    .optional(),
-  secondaryAction: z
-    .object({
-      label: z.string().optional(),
-      redirect: z.object({
-        url: z.string().optional(),
-        target: RedirectTargetEnumSchema.optional(), // Optional target
-      }),
-    })
-    .optional(), // Optional secondary action
-  data: z.record(z.unknown()).optional(), // Optional data
-  redirect: z
-    .object({
-      url: z.string().optional(),
-      target: RedirectTargetEnumSchema.optional(), // Optional target
-    })
-    .optional(),
-});
+  private buildRedirect(redirect?: InAppRedirectType) {
+    if (!(redirect && redirect.url)) {
+      return undefined;
```

Comment:
Is the default undefined or null? I think the FE expected null but I might be wrong.

Reply:
undefined, it's not a default, it's the outcome of the render.